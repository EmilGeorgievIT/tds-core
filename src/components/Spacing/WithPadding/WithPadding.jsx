import React from 'react'
import PropTypes from 'prop-types'

import safeRest from '../../../utils/safeRest'
import joinClassNames from '../../../utils/joinClassNames'

import styles from '../Spacing.modules.scss'

const getClassName = (location, scale, dangerousClassName) =>
  joinClassNames(styles[`${location || 'all'}Padding-${scale}`], dangerousClassName)

const WithPadding = ({ location, scale, dangerouslyAddClassName, children, ...rest }) => (
  <div {...safeRest(rest)} className={getClassName(location, scale, dangerouslyAddClassName)}>
    {children}
  </div>
)

WithPadding.propTypes = {
  location: PropTypes.oneOf(['horizontal', 'vertical']),
  scale: PropTypes.oneOf([1, 2, 3, 4, 6]).isRequired,
  dangerouslyAddClassName: PropTypes.string,
  children: PropTypes.node.isRequired,
}

WithPadding.defaultProps = {
  location: undefined,
  dangerouslyAddClassName: undefined,
}

export default WithPadding
