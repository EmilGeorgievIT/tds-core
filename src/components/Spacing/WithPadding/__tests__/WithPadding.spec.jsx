import React from 'react'
import { shallow, render } from 'enzyme'
import toJson from 'enzyme-to-json'

import WithPadding from '../WithPadding'

describe('WithPadding', () => {
  const defaultProps = {
    scale: 3,
  }

  const doShallow = (props = {}) =>
    shallow(
      <WithPadding {...defaultProps} {...props}>
        Some content
      </WithPadding>
    )

  const doRender = (props = {}) =>
    render(
      <WithPadding {...defaultProps} {...props}>
        Some content
      </WithPadding>
    )

  it('renders', () => {
    const withPadding = doRender()

    expect(toJson(withPadding)).toMatchSnapshot()
  })

  it('defaults to padding on all sides', () => {
    const withPadding = doShallow({ scale: 1 })

    expect(withPadding).toHaveClassName('allPadding-1')
  })

  it('can have padding', () => {
    const withPadding = doShallow({ location: 'horizontal', scale: 2 })

    expect(withPadding).toHaveClassName('horizontalPadding-2')
  })

  it('will add additional arbitrary class names', () => {
    const withPadding = doShallow({ dangerouslyAddClassName: 'a-class' })

    expect(withPadding).toHaveClassName('a-class')
  })

  it('passes additional attributes to the element', () => {
    const withPadding = doShallow({ id: 'the-id', 'data-some-attr': 'some value' })

    expect(withPadding).toHaveProp('id', 'the-id')
    expect(withPadding).toHaveProp('data-some-attr', 'some value')
  })

  it('does not allow custom CSS', () => {
    const withPadding = doShallow({ className: 'my-custom-class', style: { color: 'hotpink' } })

    expect(withPadding).not.toHaveProp('className', 'my-custom-class')
    expect(withPadding).not.toHaveProp('style')
  })
})
