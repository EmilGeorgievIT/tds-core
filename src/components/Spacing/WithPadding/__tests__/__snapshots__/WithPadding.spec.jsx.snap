// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithPadding renders 1`] = `
<div
  class="allPadding-3"
>
  Some content
</div>
`;
